linear_operator/test/linear_operator_core_test_case.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python3
+
+from abc import abstractmethod
+
+import torch
+
+import linear_operator
+from linear_operator.operators import DiagLinearOperator, to_dense
+from .base_test_case import BaseTestCase
+
+"""
+From the project description, a LinearOperator is a class that
+
+specifies the tensor(s) needed to define the LinearOperator,
+specifies a _matmul function (how the LinearOperator is applied to a vector),
+specifies a _size function (how big is the LinearOperator if it is represented as a matrix, or batch of matrices), and
+specifies a _transpose_nonbatch function (the adjoint of the LinearOperator).
+(optionally) defines other functions (e.g. logdet, eigh, etc.) to accelerate computations for which efficient
+sturcture-exploiting routines exist.
+
+What follows is a class to test these core LinearOperator operations
+"""
+
```

Comment:
This is copied from linear_operator_test_case.py. The commented out sections represent non-core operations that I dropped. I think these should be optional operations but I may have misunderstood some tests.

Reply:
I'm a fan of this! Could you refactor the current LO test case to inherit from the core case?

Reply:
Will do!